packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.spec.tsx

```diff
@@ -264,6 +264,53 @@ describe('<ArrayInput />', () => {
         expect(screen.queryByText('test helper text')).not.toBeNull();
     });
 
+    it('should update the form state to dirty, and allow submit, on updating an array input with default value', async () => {
+        const TestArrayInputs = () => {
+            const { reset } = useFormContext();
+
+            // mimic's the scaffolding elsewhere in the app but does not disable the form
+            // without this (before array input side effects)
+            React.useEffect(() => {
+                reset();
+            }, [reset]);
```

Comment:
Some help here would be useful. The only way I can scaffold the test so the form instantiates like it does in the DOM. Without this the form isn't disabled on first load even though the logic is not changed in events after this. I presume there is some composition that normally does something similar even if it via breaking memorisation somewhere? 🤷

Reply:
Refactored and works as you suggested.